### PR TITLE
Core: Add credentials to loadTable / loadView responses

### PR DESCRIPTION
**Pull request by nastra**
This implements the Spec changes from https://github.com/apache/iceberg/pull/10722

### CODE REVIEW

File: core/src/main/java/org/apache/iceberg/rest/credentials/CredentialParser.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    gen.writeEndObject();
+  }
+
+  public static Credential fromJson(String json) {
+    return JsonUtil.parse(json, CredentialParser::fromJson);
+  }
+
+  public static Credential fromJson(JsonNode json) {
+    Preconditions.checkArgument(null != json, "Cannot parse credential from null object");
+    String prefix = JsonUtil.getString(PREFIX, json);
```
Comment (amogh-jahagirdar) [inline core/src/main/java/org/apache/iceberg/rest/credentials/CredentialParser.java:59]: I think we need a check to make sure prefix isn't empty?

File: core/src/main/java/org/apache/iceberg/rest/credentials/CredentialParser.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+  public static String toJson(Credential credential, boolean pretty) {
+    return JsonUtil.generate(gen -> toJson(credential, gen), pretty);
+  }
+
+  public static void toJson(Credential credential, JsonGenerator gen) throws IOException {
+    Preconditions.checkArgument(null != credential, "Invalid credential: null");
+
+    gen.writeStartObject();
+
+    gen.writeStringField(PREFIX, credential.prefix());
```
Comment (amogh-jahagirdar) [inline core/src/main/java/org/apache/iceberg/rest/credentials/CredentialParser.java:47]: Same as below I think we need validation that the prefix isn't empty before serializing it?

File: core/src/main/java/org/apache/iceberg/rest/responses/LoadTableResponse.java [hunk 3]
```diff
@@ -110,9 +125,19 @@ public Builder addAllConfig(Map<String, String> properties) {
       return this;
     }
 
+    public Builder addCredential(Credential credential) {
+      credentials.add(credential);
+      return this;
+    }
+
+    public Builder addAllCredentials(List<Credential> credentialsToAdd) {
```
Comment (amogh-jahagirdar) [inline core/src/main/java/org/apache/iceberg/rest/responses/LoadTableResponse.java:133]: Nit: `credentialsToAdd` -> `credentials`

Comment (amogh-jahagirdar) [reply core/src/main/java/org/apache/iceberg/rest/responses/LoadTableResponse.java:133]: Ah probably checkstyle complains about the hidden field being called credentials, nvm all good.

File: core/src/main/java/org/apache/iceberg/rest/responses/LoadTableResponse.java [hunk 4]
```diff
@@ -74,12 +83,17 @@ public Map<String, String> config() {
...
+    return credentials != null ? credentials : ImmutableList.of();
+  }
+
   @Override
   public String toString() {
     return MoreObjects.toStringHelper(this)
         .add("metadataLocation", metadataLocation)
         .add("metadata", metadata)
         .add("config", config)
+        .add("credentials", credentials)
```
Comment (amogh-jahagirdar) [inline core/src/main/java/org/apache/iceberg/rest/responses/LoadTableResponse.java:96]: I'm not sure we be adding this to the `toString`? Bit concerned if someone is inadvertently logging load table response then they'd be logging sensitive info

File: core/src/main/java/org/apache/iceberg/rest/credentials/CredentialParser.java (hunk 1 above)
Comment (nastra) [reply core/src/main/java/org/apache/iceberg/rest/credentials/CredentialParser.java:59]: hm yeah good point, we should maybe do the same for the config. I've done that and added a test

File: core/src/main/java/org/apache/iceberg/rest/responses/LoadTableResponse.java (hunk 3 above)
Comment (nastra) [reply core/src/main/java/org/apache/iceberg/rest/responses/LoadTableResponse.java:133]: yeah it's due to checkstyle

**Review by amogh-jahagirdar — APPROVED** [review]
Thanks @nastra this looks good to me!

**Pull request merged**